framework/src/problems/FEProblemBase.C

```diff
@@ -933,7 +933,10 @@ FEProblemBase::prepare(const Elem * elem, THREAD_ID tid)
 
   _nl->prepare(tid);
   _aux->prepare(tid);
-  _assembly[tid]->prepare();
+  if (!_const_jacobian || _t_step <= 1 || _app.isRestarting() || _app.isRecovering() ||
```

Comment:
This is probably the line that needs to be enhanced to avoid the segfault in Bighorn

Reply:
OK, i get it, thanks @andrsd

Reply:
I don't know how to do what you suggest, @andrsd , with keeping track of whether the Jacobian has been prepared or not.  I would like to have a boolean ```jacobian_prepared_at_least_once``` that gets initialized to ```False```, and put it in the "if" statement.  But i don't know where to set that boolean to ```True```.  I'd like to set it True after we've looped over all the elements, but i got lost trying to follow the call path through ComputeJacobianThread and ThreadedElementLoopBase.  Where should i set the boolean to ```True```?

Reply:
Maybe we need to spit `FEProblemBase::prepare` into 2 methods: `prepareResidual` and `prepareJacobian`. Would that work?

Reply:
Yea, that'd work fine, @andrsd .  But i think @friedmud 's suggestion to break this PR into 2 parts: one for Assembly and the other for FEProblemBase is a good one.  I still don't know how to implement the ```jacobian_prepared_at_least_once``` boolean.  Any tips?